caikit/runtime/http_server/http_server.py

```diff
@@ -1004,6 +1039,59 @@ def _tls_files(self) -> Iterable[_TlsFiles]:
             )
             raise ValueError() from err
 
+    def _patch_openapi_spec(self):
+        """
+        FastAPI does not have a way to dynamically add openapi defs
+        for specific paths. This means we must wait till the very end
+        to update the def values. This does allow for adding context
+        specific fields though which is beneficial.
+
+        """
+        # Parse the library name into a more human readable version
+        library_name = "FastAPI"
+        if get_config().runtime.library:
+            library_name = snake_to_upper_camel(get_config().runtime.library)
+
+        # Attempt to load in the runtime library to fetch the module's docstring
+        try:
+            imported_module = importlib.import_module(get_config().runtime.library)
```

Comment:
> Is there any chance that this will change import-time semantics?

Hmmm, I actually didn't think about that. I think it should be fine since starting the runtime server happens after task&module discovery.  I liked the idea of having the module docstr in the openapi spec but maybe it should be a config variable instead?

Reply:
I think that module discovery actually _doesn't_ happen after booting the runtime, but as part of it (I had to think about this for a while). The path is:

1. The server base asks for one or more service packages ([here](https://github.com/caikit/caikit/blob/main/caikit/runtime/server_base.py#L85))
2. The `ServicePackageFactory` generates the data model ([here](https://github.com/caikit/caikit/blob/main/caikit/runtime/service_factory.py#L160))
3. The `get_data_model` function calls `_get_cdm_from_lib` ([here](https://github.com/caikit/caikit/blob/main/caikit/runtime/utils/import_util.py#L103)) which calls `get_dynamic_module` ([here](https://github.com/caikit/caikit/blob/main/caikit/runtime/utils/import_util.py#L143))
4. `get_dynamic_module` does the import ([here](https://github.com/caikit/caikit/blob/main/caikit/runtime/utils/import_util.py#L179))

Given all that, I can comfortably say that this is fine since it is invoked in `__init__` after `super().__init__()` which is what triggers the import. I think this tangled story probably belongs in a comment though!

Also, it _might_ be worth using `caikit.runtime.utils.import_utils.get_dynamic_module` here so that we consolidate all questions/concerns about `importlib` over there.

Reply:
That makes sense and matches my surface level understanding. I'll throw this in a comment in `http_server` so people know not to move it.

> Also, it might be worth using `caikit.runtime.utils.import_utils.get_dynamic_module`

Definitely! I'll use that instead